go.mod

```diff
@@ -9,19 +9,17 @@ require (
 	github.com/gorilla/context v1.1.1 // indirect
 	github.com/gorilla/mux v1.6.2
 	github.com/imdario/mergo v0.3.7 // indirect
-	github.com/onsi/ginkgo v1.11.0 // indirect
-	github.com/onsi/gomega v1.8.1 // indirect
 	github.com/openfaas/faas v0.0.0-20191125105239-365f459b3f3a
 	github.com/openfaas/faas-provider v0.15.1
 	github.com/pkg/errors v0.8.1
-	github.com/prometheus/client_golang v0.9.2
+	github.com/prometheus/client_golang v1.0.0
 	go.uber.org/goleak v1.0.0 // indirect
-	golang.org/x/sys v0.0.0-20190920190810-ef0ce1748380 // indirect
 	google.golang.org/appengine v1.6.2 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
-	k8s.io/api v0.17.4
-	k8s.io/apimachinery v0.17.4
-	k8s.io/client-go v0.17.4
-	k8s.io/code-generator v0.17.4
+	k8s.io/api v0.18.2
```

Comment:
The change updated the Go client - we will need to check if this breaks any compatibility or removes the ability to use older versions of k8s.

Reply:
> or removes the ability to use older versions of k8s.

cert-manager continues to support k8s all the way back to version 1.11 using the 1.18 dependencies: https://testgrid.k8s.io/jetstack-cert-manager-master#ci-cert-manager-e2e-v1-11

It will mean you need to change your calls to the kubernetes client interfaces to include `context.Context` as the first argument - liggitt put together a tool for automating this if you have a lot of call-sites.